pkg/allure/result.go

```diff
@@ -300,3 +301,28 @@ func getMD5Hash(text string) string {
 	hash := md5.Sum([]byte(text))
 	return hex.EncodeToString(hash[:])
 }
+
+// getTimeSumOfSteps gets first step start time and last step end time
+func getTimeSumOfSteps(result *Result) (startTimeInt int64, endTimeInt int64) {
+	startTime := result.Steps[0].Start
```

Comment:
here will be panic if Steps array is empty

although, how are you going work with following case?

```go
func TestSome(t provider.T) {
    time.Sleep(3 * time.Second)
    t.WithNewStep(...)
}
```